ibis/backends/clickhouse/__init__.py

```diff
@@ -163,6 +163,20 @@ def do_connect(
             **kwargs,
         )
 
+    @classmethod
+    def from_dbapi_connection(cls, con: cc.driver.Client) -> Backend:
```

Comment:
The latest request from this functionality comes from @sh-rp and the dltHub team: https://github.com/ibis-project/ibis/issues/8877#issuecomment-2180759817

However, for a case like ClickHouse, I'm not 100% sure we can bridge the gap, because dltHub uses `clickhouse-driver` (community package, longer history, potentially better at scale?) and Ibis uses `clickhouse-connect` (official ClickHouse package, better HTTP support). From https://github.com/ClickHouse/clickhouse-connect/issues/91, it looks like there was a lot of consideration around unification, but, in the end, doesn't seem likely.

Reply:
Scale seems like a non-issue (e.g., https://github.com/ClickHouse/clickhouse-connect/issues/91#issuecomment-1423343917).

We switched to `clickhouse-connect` due to a bunch of data type issues and choices that didn't align well with Ibis (ClickHouse enums were completely broken due to the refusal to play nice with Python's built in `Enum` types), and at the time the maintainer was unresponsive and hadn't made a release in many months.

We're definitely not prepared to support multiple ways to connect to ClickHouse right now and it's unlikely we will do so in the future.

Reply:
Makes sense! I just wanted to note this, and appreciate the explanation of why Ibis moved to `clickhouse-connect`.